lib/coverage-instrumenter.js

```diff
@@ -24,25 +25,35 @@ function getPathForRealFile(relativePath, root, templateExtensions) {
   return null;
 }
 
-function fixPath(relativePath, name, root, templateExtensions) {
+function fixPath(relativePath, name, root, templateExtensions, isAddon) {
   // Handle apps (served from app name)
-  if (relativePath.indexOf(name) === 0) {
-    relativePath = relativePath.replace(name, 'app')
-    return getPathForRealFile(relativePath, root, templateExtensions) || relativePath
+  if (!isAddon && relativePath.includes(name)) {
+    relativePath = relativePath.replace(name, 'app');
+    return getPathForRealFile(relativePath, root, templateExtensions) || relativePath;
   }
 
   // Handle addons (served from dummy app)
-  if (relativePath.indexOf('dummy') === 0) {
-    relativePath = relativePath.replace('dummy', 'app')
-    var dummyPath = path.join('tests', 'dummy', relativePath)
+  if (relativePath.includes('dummy')) {
```

Comment:
After reading the below section, it might be good to just have a top level `if (isAddon) { } else { }`, so we can keep it super clear which things apply to each thing and don't have to repeat the guard check for each "concept".

Something like:

```js
if (isAddon) {
  // Handle addon's dummy app
  if (relativePath.indexOf('dummy') === 0) {
    relativePath = relativePath.replace('dummy', 'app');
    var dummyPath = path.join('tests', 'dummy', relativePath);
    return (
      getPathForRealFile(dummyPath, root, templateExtensions) ||
      getPathForRealFile(relativePath, root, templateExtensions) ||
      relativePath
    );
  }

  // Handle addons renaming of modules/my-addon
  if (relativePath.indexOf('modules/' + name) === 0) {
    var regex = new RegExp('modules/' + name, 'g');
    relativePath = relativePath.replace(regex, 'addon');
    return (
      getPathForRealFile(relativePath, root, templateExtensions) ||
      relativePath
    );
  }

  return relativePath;
} else {
  relativePath = relativePath.replace(name, 'app');
  return getPathForRealFile(relativePath, root, templateExtensions) || relativePath;
}
```